source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizerTests/ADCIOSVisualizerTests.mm

```diff
@@ -108,10 +121,231 @@ - (void)testChoiceSetInputCanGatherDefaultValues {
     }
 }
 
-- (void)testPerformanceExample {
+- (void)testPerformanceOnComplicatedCards {
```

Comment:
>testPerformanceOnComplicatedCards [](start = 8, length = 33)

Since all of these performance test methods are identical except for the list of payload names, it may be worthwhile to factor the common code into its own method that takes the payload list as an argument.

Reply:
refactoring the measureBlock has one con. Xcode overlays test results and stats on the mesaureBlock. If mesaureBlock is refactored, then only the last run test stats is shown. I will investigate how to fix that, and update the test in the future.

---
In reply to: [219909885](https://github.com/Microsoft/AdaptiveCards/pull/1944#discussion_r219909885) [](ancestors = 219909885)